frontend/components/product-list/sections/FilterableProductsSection/useFilteredRefinementList.tsx

```diff
@@ -0,0 +1,20 @@
+import {
+   useHits,
+   useRefinementList,
+   UseRefinementListProps,
+} from 'react-instantsearch-hooks-web';
+
+export function useFilteredRefinementList(props: UseRefinementListProps) {
+   const { items, ...rest } = useRefinementList({ ...props });
+   const { results } = useHits();
+   const hitsCount = results?.nbHits ?? 0;
+
+   const filteredItems = items.reduce((acc: typeof items, item) => {
+      if (item.isRefined || hitsCount > item.count) {
+         acc.push(item);
+      }
+      return acc;
+   }, []);
```

Comment:
I think this should work equivalently.
```suggestion
   const filteredItems = items.filter((item) => item.isRefined || hitsCount > item.count);
```